buildSrc/src/integTest/groovy/org/elasticsearch/gradle/RestResourcesPluginFuncTest.groovy

```diff
@@ -0,0 +1,185 @@
+
+import org.elasticsearch.gradle.fixtures.AbstractGradleFuncTest
+import org.gradle.testkit.runner.TaskOutcome
+
+class RestResourcesPluginFuncTest extends AbstractGradleFuncTest {
+
+    def "restResources does nothing when there are no tests"() {
+        given:
+        internalBuild()
+        buildFile << """
+            apply plugin: 'elasticsearch.rest-resources'
+            apply plugin: 'elasticsearch.java'
+        """
+
+        String api = "foo.json"
+        setupRestResources([api])
+
+        when:
+        def result = gradleRunner("copyRestApiSpecsTask").build()
+
+        then:
+        result.task(':copyRestApiSpecsTask').outcome == TaskOutcome.NO_SOURCE
+        result.task(':copyYamlTestsTask').outcome == TaskOutcome.NO_SOURCE
+    }
+
+    def "restResources copies API by default for projects with tests"() {
+        given:
+        internalBuild()
+        buildFile << """
+           apply plugin: 'elasticsearch.rest-resources'
+           apply plugin: 'elasticsearch.java'
+        """
+        String api = "foo.json"
+        setupRestResources([api])
+        addRestTestsToProject(["10_basic.yml"])
+
+        when:
+        def result = gradleRunner("copyRestApiSpecsTask").build()
+
+        then:
+        result.task(':copyRestApiSpecsTask').outcome == TaskOutcome.SUCCESS
+        result.task(':copyYamlTestsTask').outcome == TaskOutcome.NO_SOURCE
+        File resourceDir = new File(testProjectDir.root, "build/resources/test/rest-api-spec")
+        new File(resourceDir, "/api/" + api).exists()
+    }
+
+    def "restResources copies all core API (but not x-pack) by default for projects with copied tests"() {
+        given:
+        internalBuild()
+        buildFile << """
+            apply plugin: 'elasticsearch.rest-resources'
+            apply plugin: 'elasticsearch.java'
+
+            restResources {
+                restTests {
+                    includeCore 'foo'
+                    includeXpack 'bar'
+                }
+            }
+        """
+        String apiCore1 = "foo1.json"
+        String apiCore2 = "foo2.json"
+        String apiXpack = "xpack.json"
+        String coreTest = "foo/10_basic.yml"
+        String xpackTest = "bar/10_basic.yml"
+        setupRestResources([apiCore1, apiCore2], [coreTest], [apiXpack], [xpackTest])
+        // intentionally not adding tests to project, they will be copied over via the plugin
+        // this tests that the test copy happens before the api copy since the api copy will only trigger if there are tests in the project
+
+        when:
+        def result = gradleRunner("copyRestApiSpecsTask").build()
+
+        then:
+        result.task(':copyRestApiSpecsTask').outcome == TaskOutcome.SUCCESS
+        result.task(':copyYamlTestsTask').outcome == TaskOutcome.SUCCESS
+        File resourceDir = new File(testProjectDir.root, "build/resources/test/rest-api-spec")
+        new File(resourceDir, "/api/" + apiCore1).exists()
```

Comment:
just a little hint. 

you could use the convenience `file()` method for better reading:
```
file("/build/resources/test/rest-api-spec/api/" + apiCore1).exists()
```
even though there's some repeating I personally think this makes the test more readable.

Reply:
done https://github.com/elastic/elasticsearch/pull/64186/commits/e02be1a384dbb1e25596fbfd5c5f48b0af36d1d0